pkg/ruler/compat.go

```diff
@@ -74,3 +78,40 @@ func engineQueryFunc(engine *promql.Engine, q storage.Queryable, delay time.Dura
 		return orig(ctx, qs, t.Add(-delay))
 	}
 }
+
+type ManagerFactory = func(
```

Comment:
This is a type alias, instead of new type (not sure it makes any difference here, but if it was a deliberate choice, I'd like to understand why?).
It should also go to ruler.go, where it is used (consumed).